filer/models/filemodels.py

```diff
@@ -506,16 +511,43 @@ def __lt__(self, other):
 
     @property
     def actual_name(self):
+        clean_name = self.clean_actual_name
+        if not self.sha1:
+            try:
+                self.generate_sha1()
+            except Exception:
```

Comment:
From examining generate_sha1 the error types that could get raised are
- IOError: based on file IO operations
- TypeError: based of bad input to the hashlib sha1 hexdigest

In any situation, is *actual_name*'s  still correct when returning the filename without the hashcode if an error occurs? The URL's  that will get generated will be wrong.